### PR TITLE
nx: simplify shared component directory structure

### PR DESCRIPTION
**Pull request by marcellmueller**
#1824

Changes shared components directory structure from `/bciers/libs/shared/components/src/lib` to `/bciers/libs/components/src`. All other shared projects are in `/bciers/libs/*` now.

To test this just run the app with `yarn dev-all` and make sure everything is working, as well as quickly check `yarn reg1` for registration 1 though as long as this builds and passes CI it will be good.

### CODE REVIEW

File: bciers/apps/registration/vitest.config.mts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default defineConfig({
       provider: "v8",
     },
     setupFiles: [
-      "../../libs/shared/testConfig/src/global.tsx",
+      "../../libs//testConfig/src/global.tsx",
```
Comment (BCerki) [inline bciers/apps/registration/vitest.config.mts:27]: There's an extra slash after `libs` in a few places

File: bciers/libs/types/.eslintrc.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
-  "extends": ["plugin:@nx/react", "../../../.eslintrc.json"],
+  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
   "ignorePatterns": ["!**/*"],
   "parserOptions": {
-    "project": ["libs/shared/actions/tsconfig(.*)?.json"],
+    "project": ["libs//types/tsconfig(.*)?.json"],
```
Comment (BCerki) [inline bciers/libs/types/.eslintrc.json:5]: I think is just a github change display thing, but is it right that this was previously actions and is now types?

**Comment by BCerki** [comment]
> All other shared projects are in `/bciers/libs/*` now.

What does `projects` mean?

**Comment by marcellmueller** [comment]
> > All other shared projects are in `/bciers/libs/*` now.
> 
> What does `projects` mean?

Any `nx project` that was in `bciers/libs/shared/*` ie `actions`, `middlewares` etc.

**Review by BCerki — COMMENTED** [review]
We still have `src/libs` in some of the other projects (e.g. actions), were you wanting to clean it up there too?

File: bciers/apps/registration/vitest.config.mts (hunk 1 above)
Comment (marcellmueller) [reply bciers/apps/registration/vitest.config.mts]: Haha thanks for catching these, it was more than just a few. I must have messed up my find and replace 😅

File: bciers/libs/types/.eslintrc.json (hunk 2 above)
Comment (marcellmueller) [reply bciers/libs/types/.eslintrc.json]: Hmm yeah I think this is some weird issue with how things were renamed. Looking at both `types` and `actions` eslint in this branch they are pointed at the correct `tsconfig` for each project.

**Review by BCerki — APPROVED** [review]
Just needs to pass CI

**Pull request merged**